Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/CertificatePersonIdentifier.kt

```diff
@@ -1,26 +1,64 @@
 package de.rki.coronawarnapp.covidcertificate.common.certificate
 
-import com.google.gson.annotations.SerializedName
+import androidx.annotation.VisibleForTesting
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
 import de.rki.coronawarnapp.covidcertificate.common.exception.InvalidHealthCertificateException.ErrorCode.DOB_MISMATCH
 import de.rki.coronawarnapp.covidcertificate.common.exception.InvalidHealthCertificateException.ErrorCode.NAME_MISMATCH
 import de.rki.coronawarnapp.covidcertificate.common.exception.InvalidVaccinationCertificateException
 import de.rki.coronawarnapp.util.HashExtensions.toSHA256
 import timber.log.Timber
 
 data class CertificatePersonIdentifier(
-    @SerializedName("dateOfBirth") val dateOfBirthFormatted: String,
-    @SerializedName("familyNameStandardized") val lastNameStandardized: String,
-    @SerializedName("givenNameStandardized") val firstNameStandardized: String?
+    @JsonProperty("dateOfBirth") val dateOfBirthFormatted: String,
+    @JsonProperty("familyNameStandardized") val lastNameStandardized: String,
+    @JsonProperty("givenNameStandardized") val firstNameStandardized: String? = null,
 ) {
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    @JsonIgnore val sanitizedFamilyName: List<String> = lastNameStandardized.sanitizeName()
+    @JsonIgnore private val sanitizedGivenName: List<String> = firstNameStandardized?.sanitizeName() ?: emptyList()
+
+    /**
+     Method shall decide whether the DGCs belong to the same holder.
+     Two DCCs shall be considered as belonging to the same holder, if:
+
+     - the sanitized `dob` attributes are the same strings, and
+     - one of:
+     - the intersection/overlap of the name components of sanitized `a.nam.fnt` and `b.nam.fnt` has at least one
+     element, and the intersection/overlap of the name components of sanitized `a.nam.gnt` and `b.nam.gnt` has at least
+     one element or both are empty sets (`gnt` is an optional field)
+     - the intersection/overlap of the name components of sanitized `a.nam.fnt` and `b.nam.gnt` has at least one
+     element, and the intersection/overlap of the name components of sanitized `a.nam.gnt` and `b.nam.fnt` has at least
+     one element
+     */
+    fun belongsToSamePerson(other: CwaCovidCertificate): Boolean = belongsToSamePerson(other.personIdentifier)
+    fun belongsToSamePerson(other: CertificatePersonIdentifier?): Boolean {
+        if (other == null) return false
+        if (dateOfBirthFormatted.trim() != other.dateOfBirthFormatted.trim()) return false
+
+        val familyNameOverlap = sanitizedFamilyName.intersect(other.sanitizedFamilyName).isNotEmpty()
```

Comment:
Can `.contains()` be used instead of `intersect().isNotEmpty()`?

If not, maybe we could create an extension function `hasIntersections()` so we don't have to call `intersect().isNotEmpty()` all the time ...

Reply:
just using `contains()` is not possible but I have created extension function `hasIntersections()`, good idea :)